Modules/TRD/src/TrackletPerTriggerCheck.cxx

```diff
@@ -0,0 +1,141 @@
+    ILOG(Debug, Support) << "configure() : using json Upperthreshold" << DesiredMeanRegion.second << ENDM;
+  } else {
+    DesiredMeanRegion.second = 520.0;
+    ILOG(Debug, Support) << "configure() : using default Upperthreshold" << DesiredMeanRegion.second << ENDM;
+  }
+
+  if (auto param = mCustomParameters.find("StatThreshold"); param != mCustomParameters.end()) {
+    StatThreshold = stod(param->second);
+    ILOG(Debug, Support) << "configure() : using json StatThreshold" << StatThreshold << ENDM;
+  } else {
+    StatThreshold = 1000;
+    ILOG(Debug, Support) << "configure() : using default StatThreshold" << StatThreshold << ENDM;
+  }
+}
+
+Quality TrackletPerTriggerCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result = Quality::Null;
+
+  // you can get details about the activity via the object mActivity:
+  ILOG(Debug, Devel) << "Run " << getActivity()->mId << ", type: " << getActivity()->mType << ", beam: " << getActivity()->mBeamType << ENDM;
+  // and you can get your custom parameters:
+  ILOG(Debug, Devel) << "custom param physics.pp.myOwnKey1 : " << mCustomParameters.atOrDefaultValue("myOwnKey1", "physics", "pp", "default_value") << ENDM;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "trackletsperevent") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+
+      // writing text over hist
+      TPaveText* msg1 = new TPaveText(0.3, 0.7, 0.7, 0.9, "NDC"); // check option "br","NDC"
+      h->GetListOfFunctions()->Add(msg1);
+      msg1->SetTextSize(10);
+      int Entries = h->GetEntries();
+      if (Entries > StatThreshold) {
+        msg1->AddText(TString::Format("Hist Can't be ignored. Stat is enough. Entries: %d > Threshold: %d", Entries, StatThreshold));
+        // msg1->SetTextColor(kGreen);
+      } else if (Entries > 0) {
+        msg1->AddText(TString::Format("Hist Can be ignored. Stat is low. Entries: %d < Threshold: %d", Entries, StatThreshold));
+        // msg1->SetTextColor(kYellow);
+      } else if (Entries == 0) {
+        msg1->AddText(TString::Format("Hist is empty. Entries: %d < Threshold: %d", Entries, StatThreshold));
+        msg1->SetTextColor(kRed);
+      }
+
+      // applying check
+      float MeanTracletPertrigger = h->GetMean();
+      if (MeanTracletPertrigger > DesiredMeanRegion.second && MeanTracletPertrigger < DesiredMeanRegion.first) {
+        result = Quality::Good;
+      } else {
+        result = Quality::Bad;
+        result.addReason(FlagReasonFactory::Unknown(), "MeanTracletPertrigger is not in bound region");
+      }
+
+      result.addMetadata("mykey", "myvalue");
+    }
+  }
+  return result;
+}
+
+std::string TrackletPerTriggerCheck::getAcceptedType() { return "TH1"; }
+
+void TrackletPerTriggerCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResult)
+{
+  if (mo->getName() == "trackletsperevent") {
+    auto* h = dynamic_cast<TH1F*>(mo->getObject());
```

Comment:
```suggestion
    auto* h = dynamic_cast<TH1F*>(mo->getObject());
```
please add a check if object was successfully casted into TH1F (is not a nullptr)